### PR TITLE
✨ Implement FavoriteFunction

### PR DESCRIPTION
**Pull request by Sina-Sche**
- [x] write customHooks (useFavorites + useStorage)

- [x]  save Favorites in LocalStorage

- [x]  display Favorites on FavoritePage

### CODE REVIEW

**Review by mhstrkmp — APPROVED** [review]
lgtm 👏

File: client/src/pages/FavoritePage.js [hunk 1]
```diff
@@ -1,15 +1,42 @@
...
+          return (
+            <InfoBox
+              key={product.id}
+              id={product.id}
+              size={"small"}
+              src={product.image}
+              alt={product.title}
+              title={product.display_title}
+              description={product.description}
+              price={product.price}
```
Comment (domko95) [inline client/src/pages/FavoritePage.js:36]:
```suggestion
              key={product.id}
              size={"small"}
              src={product.image}
              alt={product.title}
              title={product.display_title}
              {...product}
```

Not quite sure, if you need to add ```id={product.id}``` to the suggestion, but you could use the suggestion to shorten it a bit.

File: client/src/pages/SearchPage.js [hunk 2]
```diff
@@ -44,18 +44,16 @@ export const SearchPage = () => {
...
-              </Link>
+              <InfoBox
+                key={product.id}
+                id={product.id}
+                size={"small"}
+                src={product.image}
+                alt={product.title}
+                title={product.display_title}
+                description={product.description}
+                price={product.price}
```
Comment (domko95) [inline client/src/pages/SearchPage.js:55]: Same suggestion as above would work here as well.

**Review by domko95 — COMMENTED** [review]
🚀 LGTM 🚀

**Review by phlgr — APPROVED** [review]
LGTM 🚀 
Please implement suggested changes by other reviewers

File: client/src/pages/FavoritePage.js (hunk 1 above)
Comment (Sina-Sche) [reply client/src/pages/FavoritePage.js:36]: Super, danke :) Hab ich so übernommen und konnte ich auch noch auf ein paar anderen Seiten anpassen :)

Comment (Sina-Sche) [reply client/src/pages/FavoritePage.js:36]: Great! Thanks! I updated this and made the same change on some other pages as well!

**Pull request merged**